frontend/components/SetupPage/Create/SetupCreateSafe.tsx

```diff
@@ -22,47 +22,92 @@ const capitalizedMiddlewareChainNames = {
   [+MiddlewareChain.OPTIMISM]: 'Optimism',
 };
 
+const YouWillBeRedirected = ({ text }: { text: string }) => (
+  <>
+    <Image src="/onboarding-robot.svg" alt="logo" width={80} height={80} />
+    <Typography.Title
+      level={4}
+      className="m-0 mt-12 loading-ellipses"
+      style={{ width: '220px' }}
+    >
+      {text}
+    </Typography.Title>
+    <Text type="secondary">
+      You will be redirected once the account is created.
+    </Text>
+  </>
+);
+
+const CreationError = () => (
+  <>
+    <Image src="/broken-robot.svg" alt="logo" width={80} height={80} />
+    <Text type="secondary" className="mt-12">
+      Error, please restart the app and try again.
+    </Text>
+    <Text style={{ fontSize: 'small' }}>
+      If the issue persists,{' '}
+      <a href={SUPPORT_URL} target="_blank" rel="noreferrer">
+        contact Olas community support {UNICODE_SYMBOLS.EXTERNAL_LINK}
+      </a>
+      .
+    </Text>
+  </>
+);
+
 export const SetupCreateSafe = () => {
   const { goto } = usePageState();
-  const {
-    updateWallets,
-    masterSafeAddressKeyExistsForChain,
-    masterSafeAddress,
-  } = useMasterWalletContext();
-  const { updateMasterSafeOwners } = useMultisig();
+  const { masterSafes, refetch: updateWallets } = useMasterWalletContext();
+  // const { updateMasterSafeOwners } = useMultisig();
   const { backupSigner } = useSetup();
 
+  const masterSafeAddress = useMemo(() => {
+    if (!masterSafes) return;
+    const master = masterSafes.find(
+      (safe) => safe.owner === WalletOwnerType.Master,
+    );
```

Comment:
masterSafes is pre-filtered
```suggestion
    const master = masterSafes[0];
```